### PR TITLE
debug: Add ability to stop test code and resume it from SUT

### PR DESCRIPTION
**Pull request by asmorodskyi**
Add ability to stop test code and resume it from SUT

### CODE REVIEW

File: lib/utils.pm [hunk 1]
```diff
@@ -1348,4 +1348,15 @@ sub file_content_replace {
     script_run("cat $filename") if $debug;
 }
 
+=head2
+  allow to set break point in test code which could be interrupted from within SUT.
```
Comment (Martchus) [inline lib/utils.pm]: Sets a break point to pause the test execution. The test execution can be continues by creating a file like `/tmp/break0` inside the SUT.

**Comment by cfconrad** [comment]
Maybe you should use the `pool` directory. And then looking for a file called breackpoint_continue. ANd if it exists, continue but also delete that file.

File: lib/utils.pm [hunk 2]
```diff
@@ -1348,4 +1348,15 @@ sub file_content_replace {
...
+=cut
+sub conditional_sleep {
+    my ($breakpoint_number) = @_;
+    return if get_var('CONDITIONAL_SLEEP');
```
Comment (Martchus) [inline lib/utils.pm]: `return undef unless get_var('CONDITIONAL_SLEEP_ENABLED');`

**Comment by Martchus** [comment]
And the title confuses me. I guess the file needs to be created on the worker host.

**Comment by asmorodskyi** [comment]
> And the title confuses me. I guess the file needs to be created on the worker host.

actually it was not the title but code who confuses you :) idea to do this in SUT actually

**Comment by asmorodskyi** [comment]
> I think this works in a local environment. But when the environment is used by many ppl, these ppl need to sync on the number of the breakpoint.
> Its simple way of debugging. But I would not use it as openqa state of the art.

yeah you right , but WDYT about new version ?

**Comment by Martchus** [comment]
s/can be continues/can be continue/

(And yes, the typo was also in my suggested comment.)

**Comment by asmorodskyi** [comment]
> Quite hacky - this could be implemented nicely on the os-autoinst side taking advantage of the developer mode. But if you need an immediate implementation, why not.

agree , and would be happy to use such functionality as a part of developer mode . but for me it does not look like trivial thing and I think it will take some time to implement and get it into workable shape ... 
While this approach does not require too much work to use it

**Comment by cfconrad** [comment]
I think it suites more.
But Still I would delete that file, afterwards. So if I create a loop like:
```
while (true){
     if (!danger_code()){
         conditional_sleep('LOOP');
     }
}
```
It waits with each loop.

**Pull request closed without merging**